### PR TITLE
[FileProvider] Update for Xcode14 binding

### PR DESCRIPTION
**Pull request by tj-devel709**
I introduced a breaking change to the FileProvider Xcode14 binding from adding a new abstract method to the NSFileProviderPendingSetEnumerator protocol. Instead, add this method to be a Xamcore_5_0 addition.

PR where issue was introduced found here: https://github.com/xamarin/xamarin-macios/pull/15777#issuecomment-1228664877

### CODE REVIEW

File: tests/xtro-sharpie/api-annotations-dotnet/common-FileProvider.ignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# New abstract protocol member was added in Xcode14
+# which is a breaking change so adding it to XAMCORE_5_0
+!incorrect-protocol-member! NSFileProviderPendingSetEnumerator::isMaximumSizeReached is REQUIRED and should be abstract
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/api-annotations-dotnet/common-FileProvider.ignore:3]: Missing eol at the end

File: tests/xtro-sharpie/common-FileProvider.ignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# New abstract protocol member was added in Xcode14
+# which is a breaking change so adding it to XAMCORE_5_0
+!incorrect-protocol-member! NSFileProviderPendingSetEnumerator::isMaximumSizeReached is REQUIRED and should be abstract
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/common-FileProvider.ignore:3]: Same

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: tests/xtro-sharpie/api-annotations-dotnet/common-FileProvider.ignore (hunk 1 above)
Comment (dalexsoto) [inline tests/xtro-sharpie/api-annotations-dotnet/common-FileProvider.ignore:3]:
```suggestion
!incorrect-protocol-member! NSFileProviderPendingSetEnumerator::isMaximumSizeReached is REQUIRED and should be abstract

```

File: tests/xtro-sharpie/common-FileProvider.ignore (hunk 2 above)
Comment (dalexsoto) [inline tests/xtro-sharpie/common-FileProvider.ignore:3]:
```suggestion
!incorrect-protocol-member! NSFileProviderPendingSetEnumerator::isMaximumSizeReached is REQUIRED and should be abstract

```

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**